src/util/SimulationUtils.cpp

```diff
@@ -0,0 +1,284 @@
+        break;
+    case PATH_PAYMENT_STRICT_RECEIVE:
+        SimulationUtils::replaceIssuer(
+            op.body.pathPaymentStrictReceiveOp().sendAsset, n);
+        SimulationUtils::replaceIssuer(
+            op.body.pathPaymentStrictReceiveOp().destAsset, n);
+        SimulationUtils::updateAccountID(
+            op.body.pathPaymentStrictReceiveOp().destination, n);
+        for (auto& asset : op.body.pathPaymentStrictReceiveOp().path)
+        {
+            SimulationUtils::replaceIssuer(asset, n);
+        }
+        break;
+    case PATH_PAYMENT_STRICT_SEND:
+        SimulationUtils::replaceIssuer(
+            op.body.pathPaymentStrictSendOp().sendAsset, n);
+        SimulationUtils::replaceIssuer(
+            op.body.pathPaymentStrictSendOp().destAsset, n);
+        SimulationUtils::updateAccountID(
+            op.body.pathPaymentStrictSendOp().destination, n);
+        for (auto& asset : op.body.pathPaymentStrictSendOp().path)
+        {
+            SimulationUtils::replaceIssuer(asset, n);
+        }
+        break;
+    case MANAGE_SELL_OFFER:
+        offer = op.body.manageSellOfferOp().offerID;
+        SimulationUtils::replaceIssuer(op.body.manageSellOfferOp().selling, n);
+        SimulationUtils::replaceIssuer(op.body.manageSellOfferOp().buying, n);
+        if (offer > 0)
+        {
+            op.body.manageSellOfferOp().offerID =
+                SimulationUtils::getNewOfferID(offer, n);
+        }
+        break;
+    case CREATE_PASSIVE_SELL_OFFER:
+        SimulationUtils::replaceIssuer(
+            op.body.createPassiveSellOfferOp().selling, n);
+        SimulationUtils::replaceIssuer(
+            op.body.createPassiveSellOfferOp().buying, n);
+        break;
+    case SET_OPTIONS:
+        if (op.body.setOptionsOp().signer)
+        {
+            signer = *op.body.setOptionsOp().signer;
+            if (signer.key.type() == SIGNER_KEY_TYPE_ED25519)
+            {
+                signerKey = KeyUtils::convertKey<PublicKey>(signer.key);
+                signerKey = SimulationUtils::getNewSecret(
+                                KeyUtils::convertKey<PublicKey>(signer.key), n)
+                                .getPublicKey();
+                op.body.setOptionsOp().signer.activate() = Signer{
+                    KeyUtils::convertKey<SignerKey>(signerKey), signer.weight};
+            }
+        }
+        break;
+    case CHANGE_TRUST:
+        SimulationUtils::replaceIssuer(op.body.changeTrustOp().line, n);
+        break;
+    case ALLOW_TRUST:
+        SimulationUtils::updateAccountID(op.body.allowTrustOp().trustor, n);
+        break;
+    case ACCOUNT_MERGE:
+        SimulationUtils::updateAccountID(op.body.destination(), n);
+        break;
+    case MANAGE_BUY_OFFER:
+        offer = op.body.manageBuyOfferOp().offerID;
+        SimulationUtils::replaceIssuer(op.body.manageBuyOfferOp().selling, n);
+        SimulationUtils::replaceIssuer(op.body.manageBuyOfferOp().buying, n);
+        if (offer > 0)
```

Comment:
`getNewOfferID` throws if it sees an invalid offer if (0), whereas here this is a valid op and we want to just leave it at 0. That being said, now I'm thinking that perhaps `getNewOfferID` is unnecessarily aggressive and doesn't need to throw on 0, so I updated it to just return 0 instead of throwing.